src/sync/installation.ts

```diff
@@ -252,6 +256,16 @@ async function doProcessInstallation(app, data: BackfillMessagePayload, sentry:
 			try {
 				return await processor(logger, github, gitHubInstallationClient, jiraHost, repository, cursor, perPage);
 			} catch (err) {
+				// TODO - need a better way to manage GitHub errors globally
+				// In the event that the customer has not accepted the required permissions.
+				// We will continue to process the data per usual while omitting the tasks the app does not have access too.
+				// The GraphQL errors do not return a status so we check 403 or undefined
+				if ((err.status === 403 || err.status === undefined) && err.message?.includes("Resource not accessible by integration")) {
+					await subscription?.update({ syncWarning: `Invalid permissions for ${task} task` });
```

Comment:
I'm not too sure what's wrong with this piece of code - not even sure if the problem is actually here - but I can't see the syncWarning message for deployments when I haven't accepted permissions in staging.
<img width="465" alt="Screen Shot 2022-05-24 at 3 23 14 pm" src="https://user-images.githubusercontent.com/58801956/169955206-d71ef36c-64a6-4ff4-98fd-f589739516a4.png">
<img width="999" alt="Screen Shot 2022-05-24 at 3 08 19 pm" src="https://user-images.githubusercontent.com/58801956/169955223-33f781da-428e-4fb6-b294-116affba6117.png">
<img width="550" alt="Screen Shot 2022-05-24 at 3 10 18 pm" src="https://user-images.githubusercontent.com/58801956/169955227-010cf65b-4de7-43ee-87e0-6f5709f4bb95.png">